main.go

```diff
@@ -111,20 +111,48 @@ func (c *Controller) handleNewCSR(key string) error {
 		return nil
 	}
 
-	if pending := recentlyPendingCSRs(c.indexer); pending > maxPendingCSRs {
-		klog.Infof("ignoring all CSRs as too many recent pending CSRs seen: %d", pending)
-		return nil
+	machines, err := c.machines.List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list machines: %v", err)
 	}
 
-	parsedCSR, err := parseCSR(csr)
+	pendingMachinesCount := 0
+	for _, machine := range machines.Items {
+		if machine.Status.NodeRef == nil {
+			pendingMachinesCount++
+		}
+	}
+
+	// Since server-side certs are issued after client side is approved, we
```

Comment:
Based on this statement a machine can only have one pending CSR at a time, wouldn't then `maxPendingCSRs + pendingMachinesCount` be enough, why account for approved ones?

If were possible for machine to have more than one at a time couldn't we simplify by `maxPendingCSRs + 2*pendingMachinesCount`?

@enj @sjenning @sttts PTAL

Reply:
No.  If we're treating a machine without a nodeRef as a 'pending machine', that only works for the first CSR.  Once that is approved, the nodelink controller will fill in the noderef before the server-side csr is generated.  The number of 'pending machines' would go to zero before the server certs are approved, so the equation becomes 100 + 2* 0.  I put more details of what's happening in the jira ticket.